crates/matrix-sdk-crypto/src/store/memorystore.rs

```diff
@@ -136,6 +148,62 @@ impl MemoryStore {
     fn save_private_identity(&self, private_identity: Option<PrivateCrossSigningIdentity>) {
         *self.private_identity.write().unwrap() = private_identity;
     }
+
+    /// Look up the supplied backup version and provide the order number for
+    /// this backup, if we've seen it before. If not, return None.
+    fn get_backup_order(&self, backup_version: &str) -> Option<u64> {
+        self.backup_versions.read().unwrap().get(backup_version).copied()
+    }
+
+    /// Look up the supplied backup version and provide its order number. If
+    /// we've not seen it before, add it as a new order, larger than any
+    /// others that exist. If none exist, set its order to 1.
+    fn get_or_create_backup_order(&self, backup_version: &str) -> u64 {
+        let backup_versions = self.backup_versions.read().unwrap();
+        let stored = backup_versions.get(backup_version);
+        if let Some(stored) = stored {
+            *stored
+        } else {
+            drop(backup_versions);
+            // Take the write lock, then re-check whether the version exists, in case of a
+            // race where it was created in the meantime.
+            let mut backup_versions = self.backup_versions.write().unwrap();
+            let stored = backup_versions.get(backup_version);
+            if let Some(stored) = stored {
+                *stored
+            } else {
+                let new_order = backup_versions.values().max().cloned().unwrap_or(0) + 1;
+                backup_versions.insert(backup_version.to_owned(), new_order);
+                new_order
+            }
+        }
+    }
```

Comment:
as discussed, we don't think we should be keeping old backup versions, since this will go wrong in the case where a "newer" backup is deleted, and we roll back to the "older" backup. Instead we should only keep track of the "most recent" backup, and assign a new order whenever it changes.